### DISCUSSION TITLE
CSM Brainstorm

### DISCUSSION THREAD

**Discussion by jwildfire** [OpenRBQM/OpenRBQM.github.io#1] [Ideas]
We need to pick the first CSM module for the team to work on. Please drop ideas/screenshots/links in this chat and we will review at our next team meeting and select our first project!

**Comment by jwildfire** [comment]
In no particular order, here are some modules that are on my radar: 

- Endpoint Related
   - Lab Availability – monitor availability of biomarker endpoints
   - Endpoint Availability – monitor availability of symptom endpoints
       - Questionnaire
       - ePRO 
          - Timestamp analysis
          - Missing Data Trends
       - Outcome Distribution – Blinded analysis of outcome distribution
          - Binary outcome distribution
          - Over- & Under-Dispersion
      - Oral Drug Dispensation/Adherence
- Fraud Detection
   - Multivariate inlier and outliers analyses
   - Digit preference
   - Tests Distribution
   - Cluster Subjects - Duplicate enrollment
   - Distance to Site
- Domain-specific Deep Dive modules 
   - AE Deep Dive Module
      - AE Distribution
      - AE Underreporting
      - Time to first AE KRI
   - PD Deep Dive Module
      - PD trends by reasons  
   - Query Deep Dive Module
- Visit Distribution/Weekend Visits

Just picking one that is well documented - I'll add some links/screenshots related to Digit Preference shortly

**Comment by jwildfire** [comment]
# Digit Preference Module

Digit Preference links with a few representative screenshots

![image](https://github.com/user-attachments/assets/6aceef5c-1ae6-496a-a858-241bb217d39c)
- Gilead ggplot prototype

![image](https://github.com/user-attachments/assets/bcad3490-86a6-4884-a0d3-299eb7e635a2)
- [nipnTK R Package Vignette](https://nutriverse.io/nipnTK/articles/dp.html)
- [JMP implementation](https://www.jmp.com/support/downloads/JMPC71_documentation/Content/JMPCUserGuide/OP_C_FR_0015.htm)
- [digitTests R Package](https://koenderks.github.io/digitTests/)
- [dataquier R Package](https://gitlab.com/libreumg/dataquier)
- [Visualization to detect risks of clinical data in Risk-Based Quality Management](https://www.pharmasug.org/proceedings/china2021/DV/Pharmasug-China-2021-DV030.pdf)

**Reply by jonthegeek** [reply]
This has been extracted into #4. Please discuss there!

**Comment by jonthegeek** [comment]
If you have more ideas for things this team should work on, please continue to post them here!

**Comment by dennison-hall** [comment]
### Cluster within/across wireframe
![Cluster within copy](https://github.com/user-attachments/assets/776db733-d898-495d-aec3-401befb9bb27)
![Cluster across copy](https://github.com/user-attachments/assets/899b4864-1dad-4694-ad19-03d637e31613)

### Multivariate analysis wireframe
![Multivariate copy](https://github.com/user-attachments/assets/6add9231-13d8-460f-9e3f-3b03af32ca0c)

I am writing to share a set of prototypes with the community. These prototypes are conceptual in nature and are intended to serve as a foundation for sharing ideas and aligning on the direction of further development.

Please note that these prototypes do not imply any specific or existing implementation. They are purely illustrative and should be considered as a starting point for discussions and collaborative ideation.

**Comment by samussiah** [comment]
Back at Rho we developed a [set of statistical data checks](https://github.com/RhoInc/sas-statistical-data-checks/tree/master/Macros) largely focused on subject-level anomalies.  It's all written in SAS and applies a statistical model to detect outliers, but it can serve as a springboard for future RBQM modules.  Functionality includes:

- univariate analysis
- bivariate analysis
- longitudinal analysis
- group comparisons (variance)
- digit preference
- categorical distribution analysis
